great_expectations/data_context/data_context/context_factory.py

```diff
@@ -37,11 +37,333 @@
         EphemeralDataContext,
         FileDataContext,
     )
+    from great_expectations.data_context.store import (
+        CheckpointStore,
+        EvaluationParameterStore,
+        ExpectationsStore,
+        ProfilerStore,
+        ValidationsStore,
+    )
     from great_expectations.data_context.types.base import DataContextConfig
 
 ContextModes: TypeAlias = Literal["file", "cloud", "ephemeral"]
 
 
+class ProjectManager:
+    """Singleton class to manage projects in the global namespace."""
+
+    _project: AbstractDataContext | None
+
+    def __init__(self):
+        self._project = None
+
+    def get_project(  # noqa: PLR0913
+        self,
+        project_config: DataContextConfig | Mapping | None = None,
+        context_root_dir: PathStr | None = None,
+        project_root_dir: PathStr | None = None,
+        runtime_environment: dict | None = None,
+        cloud_base_url: str | None = None,
+        cloud_access_token: str | None = None,
+        cloud_organization_id: str | None = None,
+        cloud_mode: bool | None = None,
+        mode: ContextModes | None = None,
+    ) -> AbstractDataContext:
+        self._project = self._build_context(
+            project_config=project_config,
+            context_root_dir=context_root_dir,
+            project_root_dir=project_root_dir,
+            runtime_environment=runtime_environment,
+            cloud_base_url=cloud_base_url,
+            cloud_access_token=cloud_access_token,
+            cloud_organization_id=cloud_organization_id,
+            cloud_mode=cloud_mode,
+            mode=mode,
+        )
+        return self._project
+
+    def set_project(self, project: AbstractDataContext) -> None:
+        self._project = project
+
+    def get_expectations_store(self) -> ExpectationsStore:
```

Comment:
Nit - I think our convention is `expectation_store` (singular vs plural)

Only bringing it up of inconsistency with the context properties

Reply:
correct, however everywhere else its referred to as `expectations_store`, so i'm currently preferring consistency with the existing code. i think this should be renamed anyways since it _isnt_ an expectations store, its an expectations suite store!